ddmrp/models/purchase_order.py

```diff
@@ -80,8 +80,8 @@ def _find_buffer_link(self):
         buffer_model = self.env["stock.buffer"]
         move_model = self.env["stock.move"]
         for rec in self.filtered(lambda r: not r.buffer_ids):
-            mto_move = move_model.search(
-                [("created_purchase_line_id", "=", rec.id)], limit=1
+            mto_move = move_model.search([]).filtered(
+                lambda sm: rec.id in sm.created_purchase_line_ids.ids  # noqa B023
```

Comment:
This looks like a performance killer.

Doesn't `("created_purchase_line_ids", "=", rec.id)` work?